hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/service/DirectoryDeletingService.java

```diff
@@ -222,12 +225,122 @@ public BackgroundTaskResult call() {
           getOzoneManager().getMetadataManager().getTableLock(
               OmMetadataManagerImpl.DELETED_DIR_TABLE).writeLock().unlock();
         }
+
+        try {
+          if (remainNum > 0) {
+            expandSnapshotDirectories(remainNum);
+          }
+        } catch (Exception e) {
```

Comment:
`expandSnapshotDirectories` already has its own separate logging for something similar.

Reply:
Then it is double logging. May be `expandSnapshotDirectories` doesn't need to throw an `IOException`.

Reply:
It will throw because we still do `get` from RocksDB. But the exception on inside of `expandSnapshotDirectories` is specific for iterating through `snapDeletedDirTable`. The one outside is overall exeption. The main reason is `expandSnapshotDirectories` already las many `try` blocks, adding more will decrease the readability.